projects/common-platform-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/service/OffenceService.kt

```diff
@@ -0,0 +1,46 @@
+package uk.gov.justice.digital.hmpps.service
+
+import com.fasterxml.jackson.dataformat.csv.CsvMapper
+import com.fasterxml.jackson.dataformat.csv.CsvSchema
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Service
+import software.amazon.awssdk.services.s3.S3Client
+import software.amazon.awssdk.services.s3.model.GetObjectRequest
+import uk.gov.justice.digital.hmpps.integrations.client.OffencePriority
+import uk.gov.justice.digital.hmpps.integrations.delius.entity.DetailedOffenceRepository
+
+@Service
+class OffenceService(
+    private val detailedOffenceRepository: DetailedOffenceRepository,
+    private val s3Client: S3Client,
+    @Value("\${s3.bucket.name}") private val bucketName: String
+) {
+    val priorityMap: Map<String, Int> by lazy { readOffencePrioritiesFromS3() }
+
+    fun getOffenceHomeOfficeCodeByCJACode(cjaCode: String): String =
+        detailedOffenceRepository.findByCode(cjaCode)?.homeOfficeCode?.replace(
+            "/",
+            ""
+        ) ?: throw IllegalArgumentException("No Home Office code found for CJA code $cjaCode")
```

Comment:
Minor formatting thing.  IntelliJ's formatter sometimes wraps things a bit weird - would probably be clearer to wrap at the `?:`
```suggestion
    fun getOffenceHomeOfficeCodeByCJACode(cjaCode: String): String =
        detailedOffenceRepository.findByCode(cjaCode)?.homeOfficeCode?.replace("/", "") 
            ?: throw IllegalArgumentException("No Home Office code found for CJA code $cjaCode")
```

The formatter should leave it alone then.